### PR TITLE
Update 4-error-fstat.js

### PR DESCRIPTION
**Pull request by kzemlyak**
- Uses more undrestandable variable names
- Uses for loop instead of forEach

### CODE REVIEW

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
- Use camelCase: `fileName`
- Here is a case for .map

**Comment by kzemlyak** [comment]
> * Here is a case for .map

But if we use map in this case, wouldn't it be semantically incorrect because there will be mutation of the external variable (stats) inside map, while map is intended for immutable behavior?

**Comment by tshemsedinov** [comment]
> But if we use map in this case, wouldn't it be semantically incorrect because there will be mutation of the external variable (stats) inside map, while map is intended for immutable behavior?

You can easily do it without mutations. Reflect array of names to array of stats, we do not even need counters, `printResult`, and `isLastProcessed` here.

**Comment by kzemlyak** [comment]
But for that, I will need to use Promises to do this. Can I use them in this example?

**Comment by tshemsedinov** [comment]
```js
const promises = files.map((fileName) => fs.lstat(fileName));
Promise.allSettled(promises).then((stats) => {
  console.dir(stats);
});
```

**Comment by kzemlyak** [comment]
I just thought that it's not worth using Promises in this example since they were not originally included

**Comment by tshemsedinov** [comment]
We can refactor this even without promises, using closures, I'll provide multiple solutions a bit later

File: JavaScript/4-error-fstat.js [hunk 1]
```diff
@@ -2,25 +2,6 @@
 
 const fs = require('node:fs');
```
Comment (tshemsedinov) [inline JavaScript/4-error-fstat.js]:
```suggestion
const fs = require('node:fs').promises;
```

**Pull request merged**